clippy_lints/src/disallowed_methods.rs

```diff
@@ -55,19 +60,39 @@ declare_clippy_lint! {
     /// let mut xs = Vec::new(); // Vec::new is _not_ disallowed in the config.
     /// xs.push(123); // Vec::push is _not_ disallowed in the config.
     /// ```
+    ///
+    /// Profiles allow scoping different disallow lists:
+    /// ```toml
+    /// [disallowed-methods-profiles.forward_pass]
+    /// paths = [
+    ///     { path = "crate::devices::Buffer::copy_to_host", reason = "Forward code must not touch host buffers" }
+    /// ]
```

Comment:
(See other comments about API and more general examples)

```suggestion
    /// [profiles.single_threaded]
    /// disallowed_methods = [
    ///     { path = "std::thread::spawn", reason = "Unsafe to spawn more threads here" }
    /// ]
```